tests/unittest.py

```diff
@@ -164,21 +169,7 @@ def make_homeserver_config_obj(config: Dict[str, Any]) -> RootConfig:
     but it keeps a cache of `HomeServerConfig` instances and deepcopies them as needed,
     to avoid validating the whole configuration every time.
     """
-    cache_key = json.dumps(config)
-
-    if cache_key in _make_homeserver_config_obj_cache:
-        # Cache hit: reuse the existing instance
-        config_obj = _make_homeserver_config_obj_cache[cache_key]
-    else:
-        # Cache miss; create the actual instance
-        config_obj = HomeServerConfig()
-        config_obj.parse_config_dict(config, "", "")
-
-        # Add to the cache
-        _make_homeserver_config_obj_cache[cache_key] = config_obj
-
-    assert isinstance(config_obj, RootConfig)
-
+    config_obj = _parse_config_dict(json.dumps(config, sort_keys=True))
```

Comment:
[Freezing](https://github.com/matrix-org/synapse/blob/3b0083c92adf76daf4161908565de9e5efc08074/synapse/util/frozenutils.py#L20) the config dict into an immutabledict was over twice as slow as JSON serialization when I tested it. And created problems with the config parsing, because all our type checks expect lists instead of tuples and dicts instead of immutabledicts.